In [1]:
import pandas as pd

In [2]:
pd.set_option("display.max_columns", None)

In [3]:
air_quality_dataset = "data/air_quality.csv"

air_df = pd.read_csv(air_quality_dataset)

C:\Users\LENOVO\AppData\Local\Temp\ipykernel_13020\2965126160.py:3: DtypeWarning: Columns (6,7,8,9,10,11,12,13,14,15,16,18,19,20,21) have mixed types. Specify dtype option on import or set low_memory=False.
  air_df = pd.read_csv(air_quality_dataset)


In [4]:
air_df.shape

(5882208, 25)

In [5]:
air_df.head()

date   sitename          county   aqi pollutant    status  so2  \
0  2024-08-31 23:00      Hukou  Hsinchu County  62.0     PM2.5  Moderate  0.9   
1  2024-08-31 23:00  Zhongming   Taichung City  50.0       NaN      Good  1.6   
2  2024-08-31 23:00    Zhudong  Hsinchu County  45.0       NaN      Good  0.4   
3  2024-08-31 23:00    Hsinchu    Hsinchu City  42.0       NaN      Good  0.8   
4  2024-08-31 23:00     Toufen   Miaoli County  50.0       NaN      Good  1.0   

     co    o3 o3_8hr  pm10 pm2.5  no2  nox   no windspeed winddirec  unit  \
0  0.17  35.0   40.2  18.0  17.0  2.3  2.6  0.3       2.3       225   NaN   
1  0.32  27.9   35.1  27.0  14.0  7.6  9.3  1.6       1.1       184   NaN   
2  0.17  25.1   40.6  21.0  13.0  2.9  4.1  1.1       0.4       210   NaN   
3   0.2  30.0   35.9  19.0  10.0  4.0  4.8  0.7       1.9       239   NaN   
4  0.16  33.5   35.9  18.0  14.0  1.8  3.1  1.2       1.8       259   NaN   

  co_8hr pm2.5_avg pm10_avg so2_avg   longitude   latitude  siteid  
0    0.2      20.1     26.0     1.0  121.038869  24.900097    22.0  
1    0.2      15.3     23.0     1.0  120.641092  24.151958    31.0  
2    0.2      13.8     24.0     0.0  121.088955  24.740914    23.0  
3    0.2      13.0     26.0     1.0  120.972368  24.805636    24.0  
4    0.1      15.3     28.0     1.0  120.898693  24.696907    25.0

In [6]:
air_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5882208 entries, 0 to 5882207
Data columns (total 25 columns):
 #   Column     Dtype  
---  ------     -----  
 0   date       object 
 1   sitename   object 
 2   county     object 
 3   aqi        float64
 4   pollutant  object 
 5   status     object 
 6   so2        object 
 7   co         object 
 8   o3         object 
 9   o3_8hr     object 
 10  pm10       object 
 11  pm2.5      object 
 12  no2        object 
 13  nox        object 
 14  no         object 
 15  windspeed  object 
 16  winddirec  object 
 17  unit       float64
 18  co_8hr     object 
 19  pm2.5_avg  object 
 20  pm10_avg   object 
 21  so2_avg    object 
 22  longitude  float64
 23  latitude   float64
 24  siteid     float64
dtypes: float64(5), object(20)
memory usage: 1.1+ GB


In [7]:
air_df.isnull().sum()/5882208

date         0.000000
sitename     0.000000
county       0.000000
aqi          0.007314
pollutant    0.549966
status       0.024263
so2          0.023761
co           0.019296
o3           0.021253
o3_8hr       0.022284
pm10         0.023861
pm2.5        0.022345
no2          0.028217
nox          0.028752
no           0.028808
windspeed    0.037316
winddirec    0.037356
unit         1.000000
co_8hr       0.013809
pm2.5_avg    0.013590
pm10_avg     0.015279
so2_avg      0.106960
longitude    0.158719
latitude     0.158719
siteid       0.302517
dtype: float64

## Problemas

- Linhas nulas na coluna targuet status (142718)
- Coluna unit totalmente nula
- Coluna pollutant com 55% de valores nulos
- Várias outras colunas com valores nulos (abaixo de 31%)
- Várias colunas com tipos misturados

## Tasks

- Dropar as linhas que tem o target (coluna status) nulo
- Dropar a coluna unit (inútil)